Added documentation for .capacity() method (#5015)
Codecademy/docs: content/cpp/concepts/vectors/terms/size/Capacity/Capacity.md

---
Title: '.capacity()'
Description: 'Returns the storage space currently allocated for the vector'
Subjects:
  - 'Computer Science'
  - 'Game Development'
Tags:
  - 'Vectors'
  - 'Arrays'
  - 'Data Structures'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **.capacity()** method returns the storage space allocated for a vector. This is different from .size(). Capacity can be equal or greater than size. It is the total number of elements that a vector can hold without reallocation. This is automatically assigned at initialization.

Reviewer: ```suggestion
The **`.capacity()`** method returns the storage space currently allocated for a vector. This differs from [`.size()`](https://www.codecademy.com/resources/docs/cpp/vectors/size), which represents the number of elements currently in the vector. Capacity can be equal to or greater than size. It indicates the total number of elements the vector can hold without requiring reallocation. This capacity is automatically assigned when the vector is initialized.
```

Reviewer: Modifications:
1. the first instance of the term should be in **bold** and `backticks`
2. Reframed it for better clarity
3. added the description and link for the `.size()` method

Author: Hi,
I have a suggestion. Isn't it better to write `This value is automatically assigned when the vector is initialized` instead of `This capacity is automatically assigned when the vector is initialized`? Since I am defining `.capacity()`.

## Syntax

> The `.capacity()` method requires no parameters and can be called as follows

Reviewer: ```suggestion
The `.capacity()` method requires no parameters and can be used as:
```

```pseudo
myvector.capacity()
```

## Example

The following code initializes the vectors and returns its capacity

> ```codebyte/cpp

Reviewer: ```suggestion
```cpp
```

Reviewer: Since this is an example, we will use the `cpp` block. For ##Codebyte Example blocks, we use the `codebyte/cpp`

> // comparing vectors of two different capacity 
> 
> #include <iostream>
> #include <vector>
> 
> int main ()
> {
> // initialize vectors
> std::vector<int> vector_1={1,2,3};
> std::vector<int> vector_2;
> 
> // set some content in vector_2:
> for (int i=0; i<6; i++) vector_2.push_back(i);
> 
> //publish the capacity of each vector
> std::cout << "capacity of vector_1: " << (int) vector_1.capacity() << '\n';
> std::cout << "capacity of vector_2: " << (int) vector_2.capacity() << '\n';
>   
> return 0;
> }

Reviewer: ```suggestion
// Comparing vectors with different capacities
#include <iostream>
#include <vector>

int main ()
{
    // Initialize vectors
    std::vector<int> vector_1 = {1, 2, 3};
    std::vector<int> vector_2;

    // Add elements to vector_2
    for (int i = 0; i < 6; i++) 
        vector_2.push_back(i);

    // Print the capacity of each vector
    std::cout << "Capacity of vector_1: " << vector_1.capacity() << '\n';
    std::cout << "Capacity of vector_2: " << vector_2.capacity() << '\n';

    return 0;
}
```

Reviewer: added indentation for more readability and changed one comment

```

The output of the above code is

> ```

Reviewer: ```suggestion

```shell
```

Reviewer: use the ```shell``` block to show the output and also leave a space after every line

capacity of vector_1: 3

> capacity of vector_2: 8
> 

Reviewer: ```suggestion
capacity of vector_2: 8
```

Reviewer: no space is required there
